runtime/common/src/test/kotlin/kotlinx/serialization/json/JsonTestBase.kt

```diff
@@ -0,0 +1,94 @@
+    protected val nonstrict = Json(strictMode = false)
+
+    internal inline fun <reified T : Any> Json.stringify(value: T, useStreaming: Boolean): String {
+        val serializer = context.getOrDefault(T::class)
+        return stringify(serializer, value, useStreaming)
+    }
+
+    internal fun <T> Json.stringify(serializer: SerializationStrategy<T>, value: T, useStreaming: Boolean): String {
+        return if (useStreaming) {
+            stringify(serializer, value)
+        } else {
+            val tree = writeJson(value, serializer)
+            // kotlinx.serialization/issues/277
+            stringify(JsonElementSerializer, tree)
+        }
+    }
+
+    @ImplicitReflectionSerializer
+    inline fun <reified T : Any> Json.stringify(list: List<T>, useStreaming: Boolean): String {
+        return if (useStreaming) {
+            stringify(list)
+        } else {
+            stringify(context.getOrDefault(T::class).list, list)
+        }
+    }
+
+    @ImplicitReflectionSerializer
+    inline fun <reified K : Any, reified V : Any> Json.stringify(map: Map<K, V>, useStreaming: Boolean): String {
+        return if (useStreaming) {
+            stringify(map)
+        } else {
+            stringify((context.getOrDefault(K::class) to context.getOrDefault(V::class)).map, map)
+        }
+    }
+
+    internal inline fun <reified T : Any> Json.parse(source: String, useStreaming: Boolean): T {
+        val deserializer = context.getOrDefault(T::class)
+        return parse(deserializer, source, useStreaming)
+    }
+
+    internal fun <T> Json.parse(deserializer: DeserializationStrategy<T>, source: String, useStreaming: Boolean): T {
+        return if (useStreaming) {
+            parse(deserializer, source)
+        } else {
+            val parser = JsonReader(source)
+            val input = StreamingJsonInput(this, WriteMode.OBJ, parser)
+            val tree = input.readTree()
+            readJson(tree, deserializer)
+        }
+    }
+
+
+    @ImplicitReflectionSerializer
+    internal inline fun <reified T : Any> Json.parseList(content: String, useStreaming: Boolean): List<T> {
+        return if (useStreaming) {
+            parseList(content)
+        } else {
+            parse(context.getOrDefault(T::class).list, content, useStreaming)
+        }
+    }
+
+    @ImplicitReflectionSerializer
+    internal inline fun <reified K : Any, reified V : Any> Json.parseMap(
+        content: String,
+        useStreaming: Boolean
+    ): Map<K, V> {
+        return if (useStreaming) {
+            parseMap(content)
+        } else {
+            parse((context.getOrDefault(K::class) to context.getOrDefault(V::class)).map, content, useStreaming)
+        }
+    }
+
+    protected fun parametrizedTest(test: (Boolean) -> Unit) {
+        val streamingResult = kotlin.runCatching { test(true) }
+        val treeResult = kotlin.runCatching { test(false) }
+        assertEquals(streamingResult, treeResult)
+        streamingResult.getOrThrow() }
```

Comment:
brace formatting. I assume we can return from function value which was returned by `getOrThrow`?

Reply:
Just an idea. We can leave it as is for now.